src/runtime/recipe/recipe.ts

```diff
@@ -639,11 +650,20 @@ export class Recipe {
     return this.handleConnections.find(hc => !hc.type || !hc.name || hc.isOptional);
   }
 
-  getTypeHandleConnections(type, p) {
-    // returns the handles of type 'type' that do not belong to particle 'p'
-    return this.handleConnections.filter(c => {
-      return !c.isOptional && !c.handle && type.equals(c.type) && (c.particle !== p);
-    });
+  getTypeHandleConnections(type, p): {particle: Particle, connSpec: ConnectionSpec}[] {
+    // returns the {particle, connSpec} tuples for connections of type 'type'
+    // that do not belong to particle 'p'.
```

Comment:
This is a bad name, I think - there's no expectation that the function would *exclude* connections that belong to p.

Reply:
removed this method, and updated and reused another one that was available in recipe.